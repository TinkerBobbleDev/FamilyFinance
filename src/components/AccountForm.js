import React from "react";
import { withStyles } from "@material-ui/core/styles";
import TextField from "@material-ui/core/TextField";
import Button from "@material-ui/core/Button";
import { connect } from "react-redux";
import Paper from "@material-ui/core/Paper";

export const styles = theme => ({
  textField: {
    marginLeft: theme.spacing.unit,
    marginRight: theme.spacing.unit,
    marginTop: theme.spacing.unit,
    marginBottom: theme.spacing.unit,
    width: 250,
    padding: 10
  },
  button: {
    marginLeft: theme.spacing.unit,
    marginRight: theme.spacing.unit,
    width: 250,
    marginTop: theme.spacing.unit,
    marginBottom: theme.spacing.unit
  },
  paper: theme.mixins.gutters({
    paddingLeft: 0,
    paddingRight: 0,
    marginTop: theme.spacing.unit * 3,
    overflowX: "auto"
  })
});

export class AccountForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      name: props.account ? props.account.name : "",
      balance: props.account ? (props.account.balance / 100).toString() : "",
      error: "",
      submit_button_title: props.account ? "Update" : "Add",
      disableName: props.account ? true : false
    };
  }

  onSubmit = e => {
    e.preventDefault();

    if (!this.state.name || !this.state.balance) {
      this.setState(() => ({ error: "Please provide name and balance." }));
    } else {
      if (
        typeof this.props.accounts !== "undefined" &&
        this.props.accounts.length > 0
      ) {
        if (
          this.props.accounts.find(
            account => account.name === this.state.name
          ) === undefined
        ) {
          this.setState(() => ({ error: "" }));
          this.props.onSubmit({
            name: this.state.name,
            balance: parseFloat(this.state.balance, 10) * 100
          });
        } else {
          if (this.state.submit_button_title === "Update") {
            this.props.onSubmit({
              name: this.state.name,
              balance: parseFloat(this.state.balance, 10) * 100
            });
          } else {
            this.setState(() => ({
              name: "",
              balance: "",
              error: "This Account already exists"
            }));
          }
        }
      } else {
        this.setState(() => ({ error: "" }));
        this.props.onSubmit({
          name: this.state.name,
          balance: parseFloat(this.state.balance, 10) * 100
        });
      }
    }
  };

  onNameChange = e => {
    const name = e.target.value;
    this.setState(() => ({ name }));
  };

  onBalanceChange = e => {
    const balance = e.target.value;

    if (!balance || balance.match(/^\d{1,}(\.\d{0,2})?$/)) {
      this.setState(() => ({ balance }));
    }
  };

  render() {
    const { classes } = this.props;
    return (
      <Paper className={classes.paper} elevation={4}>
        {this.state.error && <p>{this.state.error}</p>}
        <form onSubmit={this.onSubmit}>
          <TextField
            required
            disabled={this.state.disableName}
            className={classes.textField}
            type="text"
            placeholder="name"
            value={this.state.name}
            onChange={this.onNameChange}
          />
          <TextField
            required
            className={classes.textField}
            type="text"
            placeholder="Balance"
            value={this.state.balance}
            onChange={this.onBalanceChange}
          />

          <Button
            onClick={this.onSubmit}
            className={classes.button}
            variant="raised"
            color="primary"
          >
            {this.state.submit_button_title}
          </Button>
        </form>
      </Paper>
    );
  }
}
const mapStateToProps = (state, props) => {
  return { accounts: state.accounts };
};

export default connect(mapStateToProps)(withStyles(styles)(AccountForm));
