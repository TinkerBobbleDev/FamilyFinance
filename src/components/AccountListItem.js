import React from "react";
import numeral from "numeral";
import TableRow from "@material-ui/core/TableRow";
import TableCell from "@material-ui/core/TableCell";
import IconButton from "@material-ui/core/IconButton";
import Menu from "@material-ui/core/Menu";
import MenuItem from "@material-ui/core/MenuItem";
import MoreVertIcon from "@material-ui/icons/MoreVert";
import { withRouter } from "react-router-dom";

const ITEM_HEIGHT = 48;

export class AccountListItems extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      anchorEl: null
    };
  }

  handleClick = event => {
    this.setState({ anchorEl: event.currentTarget });
  };

  handleClose = () => {
    this.setState({ anchorEl: null });
  };

  deleteAccount = () => {
    this.props.onDelete({ name: this.props.name });
  };

  editAccount = () => {
    this.props.history.push("/accounts/edit/" + this.props.name);
    this.setState({ anchorEl: null });
  };

  copyAccount = () => {
    this.props.onCopy(this.props.name);
    this.setState({ anchorEl: null });
  };

  render() {
    let open = Boolean(this.state.anchorEl);
    return (
      <TableRow key={this.props.name}>
        <TableCell>{this.props.name}</TableCell>
        <TableCell numeric>
          {numeral(this.props.balance / 100).format("â‚¬ 0,0.00")}
        </TableCell>
        <TableCell>
          <IconButton
            aria-label="More"
            aria-owns={open ? "long-menu" : null}
            aria-haspopup="true"
            onClick={this.handleClick}
          >
            <MoreVertIcon />
          </IconButton>
          <Menu
            id="long-menu"
            anchorEl={this.state.anchorEl}
            open={open}
            onClose={this.handleClose}
            PaperProps={{
              style: {
                maxHeight: ITEM_HEIGHT * 4.5,
                width: 200
              }
            }}
          >
            {/*<button onClick={this.deleteAccount}>Remove</button>
                        <Link to={`/accounts/edit/${this.props.id}`}>
                            <button >Edit</button>
                        </Link>*/}
            <MenuItem key="Edit" onClick={this.editAccount}>
              Edit
            </MenuItem>

            <MenuItem key="Remove" onClick={this.deleteAccount}>
              Remove
            </MenuItem>

            {/*<MenuItem key='Copy' onClick={this.copyAccount}>
                            Copy
                        </MenuItem>*/}
          </Menu>
        </TableCell>
      </TableRow>
    );
  }
}

export default withRouter(AccountListItems);
