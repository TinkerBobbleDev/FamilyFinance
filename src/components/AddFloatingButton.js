import React from "react";
import { withStyles } from "@material-ui/core/styles";
import AddIcon from "@material-ui/icons/Add";
import Button from "@material-ui/core/Button";
import Dialog from "@material-ui/core/Dialog";
import TransactionFrom from "./TransactionForm";
import { connect } from "react-redux";
import { startAddTransaction } from "../actions/transactions";
import { updateAccountBalance } from "../actions/accounts";
import withMobileDialog from "@material-ui/core/withMobileDialog";
import DialogContent from "@material-ui/core/DialogContent";
import DialogTitle from "@material-ui/core/DialogTitle";

export const styles = theme => ({
  floatingButton: {
    margin: 0,
    top: "auto",
    right: 20,
    bottom: 20,
    left: "auto",
    position: "fixed"
  },
  dialogContent: {
    textAlign: "center",
    maxWidth: 400
  },
  dialogTitle: {
    textAlign: "center"
  }
});

export class AddFloatingButton extends React.Component {
  state = {
    open: false
  };

  onSubmit = transaction => {
    if (transaction.type === "Transfer") {
      let firstTransaction = {
        type: transaction.type,
        description: transaction.description,
        amount: transaction.amount,
        category: transaction.category,
        date: transaction.date,
        account: transaction.transferFrom
      };
      let SecondTransaction = {
        type: transaction.type,
        description: transaction.description,
        amount: transaction.amount,
        category: transaction.category,
        date: transaction.date,
        account: transaction.transferTo
      };
      this.props.startAddTransfer(
        transaction,
        firstTransaction,
        SecondTransaction
      );
      return this.handleClose();
    }
    this.props.startAddTransaction(transaction);

    return this.handleClose();
  };

  handleClickFloatingButton = event => {
    this.setState({ open: true });
  };
  handleClose = () => {
    this.setState({ open: false });
  };
  render() {
    const { classes, fullScreen } = this.props;

    return (
      <div>
        <Button
          variant="fab"
          color="primary"
          aria-label="add"
          className={classes.floatingButton}
          onClick={this.handleClickFloatingButton}
        >
          <AddIcon />
        </Button>
        <Dialog
          fullScreen={fullScreen}
          open={this.state.open}
          onClose={this.handleClose}
          aria-labelledby="form-dialog-title"
        >
          <DialogTitle className={classes.dialogTitle} id="form-dialog-title">
            Add Transaction
          </DialogTitle>
          <DialogContent className={classes.dialogContent}>
            <TransactionFrom
              onSubmit={this.onSubmit}
              onClose={this.handleClose}
            />
          </DialogContent>
        </Dialog>
      </div>
    );
  }
}
const mapDispatchToProps = dispatch => ({
  startAddTransaction: transaction =>
    dispatch(startAddTransaction(transaction)).then(() => {
      let delta = transaction.amount;
      if (transaction.type === "Expense") {
        delta = -delta;
      }
      dispatch(updateAccountBalance(transaction.account, delta));
    }),
  startAddTransfer: (transaction, firstTransaction, secondTransaction) =>
    dispatch(startAddTransaction(transaction))
      .then(() => {
        dispatch(
          updateAccountBalance(firstTransaction.account, -transaction.amount)
        );
      })
      .then(() => {
        dispatch(
          updateAccountBalance(secondTransaction.account, transaction.amount)
        );
      })
});

export default connect(undefined, mapDispatchToProps)(
  withStyles(styles)(withMobileDialog()(AddFloatingButton))
);
