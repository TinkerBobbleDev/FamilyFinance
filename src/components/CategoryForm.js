import React from 'react';
import {withStyles} from 'material-ui/styles';
import TextField from 'material-ui/TextField';
import Button from 'material-ui/Button';
import {connect} from 'react-redux';
import Paper from 'material-ui/Paper';

const styles = theme => ({
    textField: {
        marginLeft: theme.spacing.unit,
        marginRight: theme.spacing.unit,
        marginTop: theme.spacing.unit,
        marginBottom: theme.spacing.unit,
        width: 250,
        padding: 10
    },
    button: {
        marginLeft: theme.spacing.unit,
        marginRight: theme.spacing.unit,
        width: 250,
        marginTop: theme.spacing.unit,
        marginBottom: theme.spacing.unit
    },
    paper: theme
        .mixins
        .gutters({

            paddingLeft: 0,
            paddingRight: 0,
            marginTop: theme.spacing.unit * 3,

            overflowX: 'auto'
        })
});

class CategoryForm extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            name: props.category
                ? props.category.name
                : '',
            error: '',
            submit_button_title: props.category
                ? 'Update'
                : 'Add'
        };
    };

    onSubmit = (e) => {
        e.preventDefault();
        if (!this.state.name) {
            this.setState(() => ({error: 'Please provide name'}));
        } else {
            if (typeof this.props.categories !== 'undefined' && this.props.categories.length > 0) {

                if (this.props.categories.find((category) => category.name === this.state.name) === undefined) {
                    this.setState(() => ({error: ''}));
                    this
                        .props
                        .onSubmit({name: this.state.name});
                } else {
                    this.setState(() => ({name: '', error: 'Category already exists'}));
                }
            } else {
                // First Added Category
                this.setState(() => ({error: ''}));
                this
                    .props
                    .onSubmit({
                        name: this.state.name,
                    });
            }
        }
    };

    onNameChange = (e) => {
        const name = e.target.value;
        this.setState(() => ({name}));
    };

    render() {
        const {classes} = this.props;
        return (
            <Paper className={classes.paper} elevation={4}>

                {this.state.error && <p>{this.state.error}</p>}
                <form onSubmit={this.onSubmit}>
                    <TextField
                        className={classes.textField}
                        type="text"
                        placeholder="category"
                        value={this.state.name}
                        onChange={this.onNameChange}/>

                    <Button
                        onClick={this.onSubmit}
                        className={classes.button}
                        variant="raised"
                        color="primary">{this.state.submit_button_title}</Button>
                </form>
            </Paper>
        )
    }

}

const mapStateToProps = (state, props) => {
    return {categories: state.categories};
};

export default connect(mapStateToProps)(withStyles(styles)(CategoryForm));