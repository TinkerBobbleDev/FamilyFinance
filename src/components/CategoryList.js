import React from 'react';
import {connect} from 'react-redux';
import CategoryListItem from './CategoryListItem';
import {startDeleteCategory, startAddCategory} from '../actions/categories';
import {withStyles} from 'material-ui/styles';
import Table, {TableBody, TableCell, TableHead, TableRow} from 'material-ui/Table';

const styles = theme => ({
    root: {
        width: '100%',
        marginTop: theme.spacing.unit * 3,
        overflowX: 'auto'
    },
    table: {
        minWidth: 500,
        
    }
});

export class CategoryList extends React.Component {

    onDelete = (id) => {
        this
            .props
            .startDeleteCategory(id);
    };

    onCopy = (id) => {
        console.log(id)
        this
            .props
            .categories
            .map((category) => {
                if (category.id === id) {
                    return this
                        .props
                        .startAddCategory(category);
                } else {
                    return console.log('no category with this ID found')
                }
            })

    };

    render() {
        const {classes} = this.props;
        return (
            <div>
                {this.props.categories.lenght === 0
                    ? (
                        <p>no categories</p>
                    )
                    : (
                        <Table className={classes.table}>
                            <TableHead>
                                <TableRow>
                                    <TableCell>Category</TableCell>
                                    <TableCell ></TableCell>
                                </TableRow>
                            </TableHead>
                            <TableBody>
                                { this
                                    .props
                                    .categories
                                    .map((category) => {
                                        return <CategoryListItem 
                                            key={category.id} 
                                            onDelete={this.onDelete}
                                            onCopy={this.onCopy}
                                            {...category}/>
                                    })
                                }
                            </TableBody>
                        </Table>
                    )
                }
            </div>

        )
    }
}

const mapStateToProps = (state, props) => {
    return {
        categories: state.categories,
        ...props
    };
};

const mapDispatchToProps = (dispatch, props) => ({
    startAddCategory: (category) => dispatch(startAddCategory(category)),
    startDeleteCategory: (data) => dispatch(startDeleteCategory(data))
});

export default connect(mapStateToProps, mapDispatchToProps)(withStyles(styles)(CategoryList))