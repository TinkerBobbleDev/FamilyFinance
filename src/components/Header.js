import React from 'react';
import {NavLink} from 'react-router-dom';
import classNames from 'classnames'
import AppBar from 'material-ui/AppBar';
import Avatar from 'material-ui/Avatar';
import {withStyles} from 'material-ui/styles';
import Typography from 'material-ui/Typography';
import Toolbar from 'material-ui/Toolbar';
import MenuIcon from 'material-ui-icons/Menu';
import IconButton from 'material-ui/IconButton';
import Button from 'material-ui/Button';
import {connect} from 'react-redux';
import {startLogout} from '../actions/auth';
import Constants from '../utils/constants';
import AddIcon from 'material-ui-icons/NoteAdd';
import DashboardIcon from 'material-ui-icons/Dashboard';
import ShowChartIcon from 'material-ui-icons/ShowChart';
import AccountBalanceIcon from 'material-ui-icons/AccountBalance';
import Hidden from 'material-ui/Hidden';
import {ListItem, ListItemText} from 'material-ui/List';
import List from 'material-ui/List';
import Divider from 'material-ui/Divider';
import Drawer from 'material-ui/Drawer';
import ChevronLeftIcon from 'material-ui-icons/ChevronLeft';
import ChevronRightIcon from 'material-ui-icons/ChevronRight';

const drawerWidth = 280;
const styles = theme => ({
    root: {
        width: '100%',
        height: 'auto',
        zIndex: 1,
        overflow: 'hidden'
    },
    flex: {
        flex: 1
    },
    menuButton: {
        marginLeft: 12,
        marginRight: 36
    },
    hide: {
        display: 'none'
    },

    drawerPaper: {
        position: 'fix',
        height: '100%',
        width: drawerWidth,
        transition: theme
            .transitions
            .create('width', {
                easing: theme.transitions.easing.sharp,
                duration: theme.transitions.duration.enteringScreen
            })
    },
    drawerPaperClose: {
        width: 60,
        overflowX: 'hidden',
        transition: theme
            .transitions
            .create('width', {
                easing: theme.transitions.easing.sharp,
                duration: theme.transitions.duration.leavingScreen
            })
    },
    drawerInner: {
        // Make the items inside not wrap when transitioning:
        width: drawerWidth
    },
    drawerHeader: {
        display: 'flex',
        alignItems: 'center',
        justifyContent: 'flex-end',
        padding: '0 8px',
        ...theme.mixins.toolbar
    },
    button: {
        margin: theme.spacing.unit
    },

    avatar: {
        margin: 10
    },

    appBar: {
        position: 'fixed',
        zIndex: theme.zIndex.navDrawer + 1,
        transition: theme
            .transitions
            .create([
                'width', 'margin'
            ], {
                easing: theme.transitions.easing.sharp,
                duration: theme.transitions.duration.leavingScreen
            })
    },
    appBarShift: {
        marginLeft: drawerWidth,
        width: `calc(100% - ${drawerWidth}px)`,
        transition: theme
            .transitions
            .create([
                'width', 'margin'
            ], {
                easing: theme.transitions.easing.sharp,
                duration: theme.transitions.duration.enteringScreen
            })
    },
    appFrame: {

        position: 'relative',
        display: 'flex',
        width: '100%'
    },
    // content: {
    //     width: '100%',
    //     flexGrow: 1,
    //     padding: theme.spacing.unit * 3,
    //     height: 'calc(100% - 56px)',
    //     marginTop: 26,
    //     [
    //         theme
    //             .breakpoints
    //             .up('sm')
    //     ]: {
    //         height: 'calc(100% - 64px)',
    //         marginTop: 64
    //     }
    // },
    content: {
        width: '100%',
        flexGrow: 1,
        padding: 20,
        height: 'calc(100% - 50px)',
        marginTop: 26,
        [
            theme
                .breakpoints
                .up('sm')
        ]: {
            height: 'calc(100% - 64px)',
            width: 'calc(100% - 60px)',
            marginTop: 24,
            marginBottom: 80,
            margin: 40
        },
        [theme.breakpoints.down('sm')]: {
            padding:0,
            marginTop: 44,
            marginBottom: 80
          },
    }
});

class Header extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
            open: false,
            mobileOpen: false
        }
    }
    handleDrawerToggle = () => {
        this.setState({
            mobileOpen: !this.state.mobileOpen
        });
    };

    handleDrawerOpen = () => {
        this.setState({open: true});
    };

    handleDrawerClose = () => {
        this.setState({open: false});
    };
    render() {
        const {classes, theme} = this.props;

        const items = (

            <List>
                <div>

                    <NavLink
                        to="/dashboard"
                        activeClassName="is-active"
                        activeStyle={{
                        fontWeight: 'bold',
                        color: 'red'
                    }}
                        style={{
                        textDecoration: 'none'
                    }}>
                        <ListItem button divider>
                            <ShowChartIcon/>
                            <ListItemText primary="Dashboard"/>
                        </ListItem>
                    </NavLink>
                    <NavLink
                        to="/accounts"
                        activeClassName="is-active"
                        activeStyle={{
                        fontWeight: 'bold',
                        color: 'red',
                        fontColor: 'red'
                    }}
                        style={{
                        textDecoration: 'none'
                    }}>
                        <ListItem button divider>
                            <AccountBalanceIcon/>
                            <ListItemText primary="Accounts"/>
                        </ListItem>
                    </NavLink>
                    <NavLink
                        to="/transactions"
                        activeClassName="is-active"
                        activeStyle={{
                        fontWeight: 'bold',
                        color: 'red'
                    }}
                        style={{
                        textDecoration: 'none'
                    }}>
                        <ListItem button divider>
                            <AddIcon/>
                            <ListItemText primary="Transactions"/>
                        </ListItem>
                    </NavLink>
                    <NavLink
                        to="/categories"
                        activeClassName="is-active"
                        activeStyle={{
                        fontWeight: 'bold',
                        color: 'red'
                    }}
                        style={{
                        textDecoration: 'none'
                    }}>
                        <ListItem button divider>
                            <DashboardIcon/>
                            <ListItemText primary="Categories"/>
                        </ListItem>
                    </NavLink>
                </div>
            </List>

        );

        return (
            <div className={classes.root}>
                <div className={classes.appFrame}>
                    <AppBar
                        className={classNames(classes.appBar, this.state.open && classes.appBarShift)}
                        position="static"
                        color="primary">
                        <Toolbar disableGutters={!this.state.open}>
                            <IconButton
                                color="contrast"
                                aria-label="open drawer"
                                onClick={this.handleDrawerOpen}
                                className={classNames(classes.menuButton, this.state.open && classes.hide)}>
                                <MenuIcon/>
                            </IconButton>
                            <Typography type="title" color="inherit" className={classes.flex}>
                                {this.props.title}
                            </Typography>

                            <Button onClick={this.props.startLogout} raised className={classes.button}>
                                {Constants.ASSIGNS_LOGOUT}
                            </Button>
                            <Avatar alt="Name" src={this.props.avatarSrc} className={classes.avatar}/>

                        </Toolbar>
                    </AppBar>
                    <Hidden mdUp>
                        <Drawer
                            type="temporary"
                            anchor={theme.direction === 'rtl'
                            ? 'right'
                            : 'left'}
                            open={this.state.open}
                            classes={{
                            paper: classNames(classes.drawerPaper, !this.state.open && classes.drawerPaperClose)
                        }}
                            onClose={this.handleDrawerToggle}
                            ModalProps={{
                            keepMounted: true
                        }}>
                            <div className={classes.drawerInner}>
                                <div className={classes.drawerHeader}>
                                    <IconButton onClick={this.handleDrawerClose}>
                                        {theme.direction === 'rtl'
                                            ? <ChevronRightIcon/>
                                            : <ChevronLeftIcon/>}
                                    </IconButton>
                                </div>
                                <Divider/> {items}
                            </div>
                        </Drawer>
                    </Hidden>
                    <Hidden smDown implementation="css">
                        <Drawer
                            type="permanent"
                            classes={{
                            paper: classNames(classes.drawerPaper, !this.state.open && classes.drawerPaperClose)
                        }}
                            open={this.state.open}>
                            <div className={classes.drawerInner}>
                                <div className={classes.drawerHeader}>
                                    <IconButton onClick={this.handleDrawerClose}>
                                        {theme.direction === 'rtl'
                                            ? <ChevronRightIcon/>
                                            : <ChevronLeftIcon/>}
                                    </IconButton>
                                </div>
                                <Divider/> {items}
                            </div>
                        </Drawer>
                    </Hidden>
                </div>
                <main className={classes.content}>
                    {this.props.children}
                </main>
            </div>

        )
    }
}
const mapStateToProps = (state, props) => ({avatarSrc: state.auth.photoURL});

const mapDispatchToProps = (dispatch) => ({
    startLogout: () => dispatch(startLogout())
});

export default connect(mapStateToProps, mapDispatchToProps)(withStyles(styles, {withTheme: true})(Header));