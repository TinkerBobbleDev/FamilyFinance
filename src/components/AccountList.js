import React from 'react';
import {connect} from 'react-redux';
import AccountListItem from './AccountListItem';
import {startDeleteAccount, startAddAccount} from '../actions/accounts';
import {withStyles} from 'material-ui/styles';
import Table, {TableBody, TableCell, TableHead, TableRow} from 'material-ui/Table';

const styles = theme => ({
    root: {
        width: '100%',
        marginTop: theme.spacing.unit * 3,
        overflowX: 'auto'
    },
    table: {
        minWidth: 700
    }
});

export class AccountList extends React.Component {

    onDelete = (id) => {
        this
            .props
            .startDeleteAccount(id);
    };

    onCopy = (id) => {
        console.log(id)
        this
            .props
            .accounts
            .map((account) => {
                if (account.id === id) {
                    return this
                        .props
                        .startAddAccount(account);
                } else {
                    return console.log('no account with this ID found')
                }
            })

    };

    render() {
        const {classes} = this.props;
        return (
            <div>
                {this.props.accounts.lenght === 0
                    ? (
                        <p>no accounts</p>
                    )
                    : (
                        <Table className={classes.table}>
                            <TableHead>
                                <TableRow>
                                    <TableCell>Account</TableCell>
                                    <TableCell numeric>Balance</TableCell>
                                    <TableCell ></TableCell>

                                </TableRow>
                            </TableHead>
                            <TableBody>
                                {this
                                    .props
                                    .accounts
                                    .map((account) => {
                                        return <AccountListItem 
                                            key={account.id} 
                                            onDelete={this.onDelete}
                                            onCopy={this.onCopy}
                                            {...account}/>
                                    })
                                }
                            </TableBody>
                        </Table>
                    )
                }
            </div>

        )
    }
}

const mapStateToProps = (state, props) => {
    return {
        accounts: state.accounts,
        ...props
    };
};

const mapDispatchToProps = (dispatch, props) => ({
    startAddAccount: (account) => dispatch(startAddAccount(account)),
    startDeleteAccount: (data) => dispatch(startDeleteAccount(data))
});

export default connect(mapStateToProps, mapDispatchToProps)(withStyles(styles)(AccountList))