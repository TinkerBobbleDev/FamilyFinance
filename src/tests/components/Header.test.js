import React from "react";
// import ReactShallowRenderer from "react-test-renderer/shallow";
import { Header } from "../../components/Header";
import toJSON from "enzyme-to-json";
import { createShallow } from "@material-ui/core/test-utils";
import { styles } from "../../components/Header";

describe("<Header />", () => {
  let shallow;

  beforeAll(() => {
    shallow = createShallow();
  });

  it("should work", () => {
    const wrapper = shallow(<Header theme={{}} classes={styles} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
