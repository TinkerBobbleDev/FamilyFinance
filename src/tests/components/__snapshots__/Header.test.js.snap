// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should work 1`] = `
<div>
  <div>
    <WithStyles(AppBar)
      className=""
      color="primary"
      position="static"
    >
      <WithStyles(Toolbar)
        disableGutters={true}
      >
        <WithStyles(IconButton)
          aria-label="open drawer"
          className=""
          color="inherit"
          onClick={[Function]}
        >
          <pure(Menu) />
        </WithStyles(IconButton)>
        <WithStyles(Typography)
          color="inherit"
          variant="headline"
        />
        <WithStyles(IconButton)
          aria-haspopup="true"
          aria-owns={null}
          color="inherit"
          onClick={[Function]}
        >
          <pure(AccountCircle) />
        </WithStyles(IconButton)>
        <WithStyles(Menu)
          anchorEl={null}
          anchorOrigin={
            Object {
              "horizontal": "right",
              "vertical": "top",
            }
          }
          id="menu-appbar"
          onClose={[Function]}
          open={false}
          transformOrigin={
            Object {
              "horizontal": "right",
              "vertical": "top",
            }
          }
        >
          <WithStyles(MenuItem)>
             
            <WithStyles(Avatar)
              alt="Name"
            />
          </WithStyles(MenuItem)>
          <WithStyles(MenuItem)>
            Logout
          </WithStyles(MenuItem)>
        </WithStyles(Menu)>
      </WithStyles(Toolbar)>
    </WithStyles(AppBar)>
    <Hidden
      implementation="js"
      lgDown={false}
      lgUp={false}
      mdDown={false}
      mdUp={true}
      smDown={false}
      smUp={false}
      xlDown={false}
      xlUp={false}
      xsDown={false}
      xsUp={false}
    >
      <WithStyles(Drawer)
        ModalProps={
          Object {
            "keepMounted": true,
          }
        }
        anchor="left"
        classes={
          Object {
            "paper": "",
          }
        }
        onClose={[Function]}
        open={false}
        variant="temporary"
      >
        <div>
          <div>
            <WithStyles(IconButton)
              onClick={[Function]}
            >
              <pure(ChevronLeft) />
            </WithStyles(IconButton)>
          </div>
          <WithStyles(Divider) />
           
          <WithStyles(List)>
            <div>
              <NavLink
                activeClassName="is-active"
                activeStyle={
                  Object {
                    "color": "red",
                    "fontWeight": "bold",
                  }
                }
                ariaCurrent="true"
                style={
                  Object {
                    "textDecoration": "none",
                  }
                }
                to="/dashboard"
              >
                <WithStyles(ListItem)
                  button={true}
                  divider={true}
                >
                  <pure(ShowChart) />
                  <WithStyles(ListItemText)
                    primary="Dashboard"
                  />
                </WithStyles(ListItem)>
              </NavLink>
              <NavLink
                activeClassName="is-active"
                activeStyle={
                  Object {
                    "color": "red",
                    "fontColor": "red",
                    "fontWeight": "bold",
                  }
                }
                ariaCurrent="true"
                style={
                  Object {
                    "textDecoration": "none",
                  }
                }
                to="/accounts"
              >
                <WithStyles(ListItem)
                  button={true}
                  divider={true}
                >
                  <pure(AccountBalance) />
                  <WithStyles(ListItemText)
                    primary="Accounts"
                  />
                </WithStyles(ListItem)>
              </NavLink>
              <NavLink
                activeClassName="is-active"
                activeStyle={
                  Object {
                    "color": "red",
                    "fontWeight": "bold",
                  }
                }
                ariaCurrent="true"
                style={
                  Object {
                    "textDecoration": "none",
                  }
                }
                to="/transactions"
              >
                <WithStyles(ListItem)
                  button={true}
                  divider={true}
                >
                  <pure(NoteAdd) />
                  <WithStyles(ListItemText)
                    primary="Transactions"
                  />
                </WithStyles(ListItem)>
              </NavLink>
              <NavLink
                activeClassName="is-active"
                activeStyle={
                  Object {
                    "color": "red",
                    "fontWeight": "bold",
                  }
                }
                ariaCurrent="true"
                style={
                  Object {
                    "textDecoration": "none",
                  }
                }
                to="/categories"
              >
                <WithStyles(ListItem)
                  button={true}
                  divider={true}
                >
                  <pure(Dashboard) />
                  <WithStyles(ListItemText)
                    primary="Categories"
                  />
                </WithStyles(ListItem)>
              </NavLink>
            </div>
          </WithStyles(List)>
        </div>
      </WithStyles(Drawer)>
    </Hidden>
    <Hidden
      implementation="css"
      lgDown={false}
      lgUp={false}
      mdDown={false}
      mdUp={false}
      smDown={true}
      smUp={false}
      xlDown={false}
      xlUp={false}
      xsDown={false}
      xsUp={false}
    >
      <WithStyles(Drawer)
        classes={
          Object {
            "paper": "",
          }
        }
        open={false}
        variant="permanent"
      >
        <div>
          <WithStyles(IconButton)
            onClick={[Function]}
          >
            <pure(ChevronLeft) />
          </WithStyles(IconButton)>
        </div>
        <WithStyles(Divider) />
         
        <WithStyles(List)>
          <div>
            <NavLink
              activeClassName="is-active"
              activeStyle={
                Object {
                  "color": "red",
                  "fontWeight": "bold",
                }
              }
              ariaCurrent="true"
              style={
                Object {
                  "textDecoration": "none",
                }
              }
              to="/dashboard"
            >
              <WithStyles(ListItem)
                button={true}
                divider={true}
              >
                <pure(ShowChart) />
                <WithStyles(ListItemText)
                  primary="Dashboard"
                />
              </WithStyles(ListItem)>
            </NavLink>
            <NavLink
              activeClassName="is-active"
              activeStyle={
                Object {
                  "color": "red",
                  "fontColor": "red",
                  "fontWeight": "bold",
                }
              }
              ariaCurrent="true"
              style={
                Object {
                  "textDecoration": "none",
                }
              }
              to="/accounts"
            >
              <WithStyles(ListItem)
                button={true}
                divider={true}
              >
                <pure(AccountBalance) />
                <WithStyles(ListItemText)
                  primary="Accounts"
                />
              </WithStyles(ListItem)>
            </NavLink>
            <NavLink
              activeClassName="is-active"
              activeStyle={
                Object {
                  "color": "red",
                  "fontWeight": "bold",
                }
              }
              ariaCurrent="true"
              style={
                Object {
                  "textDecoration": "none",
                }
              }
              to="/transactions"
            >
              <WithStyles(ListItem)
                button={true}
                divider={true}
              >
                <pure(NoteAdd) />
                <WithStyles(ListItemText)
                  primary="Transactions"
                />
              </WithStyles(ListItem)>
            </NavLink>
            <NavLink
              activeClassName="is-active"
              activeStyle={
                Object {
                  "color": "red",
                  "fontWeight": "bold",
                }
              }
              ariaCurrent="true"
              style={
                Object {
                  "textDecoration": "none",
                }
              }
              to="/categories"
            >
              <WithStyles(ListItem)
                button={true}
                divider={true}
              >
                <pure(Dashboard) />
                <WithStyles(ListItemText)
                  primary="Categories"
                />
              </WithStyles(ListItem)>
            </NavLink>
          </div>
        </WithStyles(List)>
      </WithStyles(Drawer)>
    </Hidden>
  </div>
  <main />
</div>
`;
